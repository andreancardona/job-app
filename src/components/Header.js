import React from 'react';
import AddJobButton from './AddJobButton';
import '../styles/header.css';


const Header = (props) => {
  return (
    <div className="header-container">
      <div className="header-jobs">
        Jobs 
        <div className="listings-count">{props.jobs.length} listings</div>
      </div>
      <AddJobButton />
    </div>
  )
}

export default Header;