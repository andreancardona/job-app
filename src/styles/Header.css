/* header */

.header-container {
  margin-top: 150px;
  width: 70%;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.header-jobs {
  font-family: "Noto Sans";
  font-style: normal;
  font-weight: 700;
  font-size: 28px;
  line-height: 35px;
  color: #2d2d2d;

  display: flex;
  flex-direction: row;
}

.listings-count {
  font-family: "Noto Sans";
  font-style: normal;
  font-weight: 300;
  font-size: 14px;
  line-height: 21px;
  color: #949494;
  margin: 11px 0 0 10px;
}

@media screen and (max-width: 768px) {
  .header-container {
    margin-top: 30%;
    width: 80%;
  }
  .header-jobs {
    font-size: 20px;
  }

  .listings-count {
    margin-top: 8px;
  }
}