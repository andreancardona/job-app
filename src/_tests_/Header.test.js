import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import Header from '../components/Header';

 // Initial list of jobs
 const jobs = [
  { id: 1, title: 'Product Manager', location: 'Austin, TX', posted: '10/04/2020', sponsorship: 'Free', status: 'Open' },
  { id: 2, title: 'CEO', location: 'Austin, TX', posted: '12/30/2020', sponsorship: 'Sponsored', status: 'Paused'},
  { id: 3, title: 'Software Engineer', location: 'Seattle, WA', posted: '11/18/2020', sponsorship: 'Free', status: 'Closed' },
]


describe('The <Header /> component', () => {
  test('Header renders with Header props', () => {
    const tree = renderer.create(<Header jobs={jobs}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('Job should not fail any accessibility tests', async () => {
    const { container } = render(<Header jobs={jobs}/>);
    expect(await axe(container)).toHaveNoViolations();
  });
});
