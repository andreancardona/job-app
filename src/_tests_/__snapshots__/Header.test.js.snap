// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Header /> component Header renders with Header props 1`] = `
<div
  className="header-container"
>
  <div
    aria-level="1"
    className="header-jobs"
    role="heading"
    tabIndex="0"
  >
    Jobs
    <div
      className="listings-count"
    >
      3
       listings
    </div>
  </div>
  <button
    className="add-save-button"
    onClick={[Function]}
  >
    Add job
  </button>
</div>
`;
